sympy/core/relational.py

```diff
@@ -1138,3 +1032,265 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+def fuzzy_resolve(lhs, rhs, op, cls):
+    """
+    Resolves op to S.true if cmp returns true.
+    or to S.false if cmp returns false
+    or to an unevaluated if cmp returns none.
+    """
+    val = op(lhs, rhs)
+    if val is None:
+        return cls(lhs, rhs, evaluate=False)
+    elif isinstance(val,bool):
+        if val is True:
+            return S.true
+        else:
+            return S.false
+    else:
+        return val
+
+
+def is_lt(lhs, rhs):
+    retval = None
+    if lhs.is_extended_real and rhs.is_extended_real and hasattr(lhs, "_eval_is_ge"):
+        retval = fuzzy_not(getattr(lhs, "_eval_is_ge")(rhs))
+    elif hasattr(lhs, "_eval_is_lt"):
+        retval = getattr(lhs, "_eval_is_lt")(rhs)
+    if retval is None:
+        retval = cmp(lhs, rhs, "<")
+    return retval
+
+
+def is_le(lhs, rhs):
+    retval = None
+    if hasattr(lhs, "_eval_is_gt"):
+        retval = fuzzy_not(getattr(lhs, "_eval_is_gt")(rhs))
+    if retval is None:
+        retval = cmp(lhs, rhs, "<=")
+    return retval
+
+
+def is_gt(lhs, rhs):
+    retval = None
+    if hasattr(lhs, "_eval_is_gt"):
+        retval = getattr(lhs, "_eval_is_gt")(rhs)
+    if retval is None:
+        retval = cmp(lhs, rhs, ">")
+    return retval
+
+
+def is_ge(lhs, rhs):
+    retval = None
+    if hasattr(lhs, "_eval_is_ge"):
+        retval = getattr(lhs, "_eval_is_ge")(rhs)
+    if retval is None:
+        retval = cmp(lhs, rhs, ">=")
+    return retval
+
+
+def is_eq(lhs, rhs):
+    retval = None
+    if hasattr(lhs, "_eval_Eq"):
+        retval = getattr(lhs, "_eval_Eq")(rhs)
+    elif hasattr(rhs, "_eval_Eq"):
+        retval = getattr(rhs, "_eval_Eq")(lhs)
+
+    if retval is None:
+        retval = _cmp_eq(lhs, rhs)
+    return retval
+
+
+def is_neq(lhs, rhs):
+    retval = None
+    if hasattr(lhs, "_eval_Eq"):
+        retval = fuzzy_not(getattr(lhs, "_eval_Eq")(rhs))
```

Comment:
Can this whole function just be `return fuzzy_not(is_eq(lhs, rhs))`?

Reply:
yes